manual/src/cmds/runtime-tracing.etex

```diff
@@ -70,6 +72,33 @@ The full set of events emitted by probes and their documentation can be found in
  section~\ref{Runtime_events}.
 \fi
 
+\subsection{s:runtime-tracing-custom-events}{Custom events}
+The runtime events system supports defining custom events. Identified by a
+globally unique name, they are emitted with payloads of built-in (\texttt{unit},
+\texttt{int}, \texttt{span}) and user-defined types. They are registered using
+the \texttt{Runtime_events.User.register} function. Once an event is registered,
+values for that event are emitted using \texttt{Runtime_events.User.write}.
+
+As consumer and producer can be different programs, they might not have the same
+set of registered events. The event name is used by the consumer to identify
+producer events, with a fallback mechanism for unknown events of built-in types.
+Unknown events for user-defined types are currently ignored, even if the type is
+defined both in the producer and consumer side.
+
+An extensible variant \texttt{tag} provides a way to attach additional data to
+events. The tag is not serialized, so it is only available for \textit{known}
+events.
```

Comment:
I don't understand what the last sentence means.

Reply:
When the consumer receives an event that it doesn't know about, only the name and payload content can be retrieved, not the tag.

Reply:
In the ring buffer events are just an index in to a name string table and a bunch of bytes.

A tag is essentially a way of you associating an OCaml type with the name, so you don't have to use the string name everywhere. If the producer and consumer both know about a custom event custom type, they can register the same tag. That's not necessary though, if you register a callback for a custom type but don't register tags, you should still get the events - they'll just identified by a name and not have a tag. For the common use case of taking these events and recording them in to a log file or database somewhere, that's sufficient.